Estadística descriptiva y probabilidades


1. El promedio general de los salarios, desviación estándar, quintiles y rango. (hint:escoge la columna adecuada para comparar).

In [103]:
import pandas as pd 
import numpy as np

df = pd.read_csv('ds_salaries.csv')
df.head(3)

work_year experience_level employment_type                 job_title  \
0       2023               SE              FT  Principal Data Scientist   
1       2023               MI              CT               ML Engineer   
2       2023               MI              CT               ML Engineer   

   salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
0   80000             EUR          85847                 ES           100   
1   30000             USD          30000                 US           100   
2   25500             USD          25500                 US           100   

  company_location company_size  
0               ES            L  
1               US            S  
2               US            S

In [104]:

# MEDIDAD DE DISPERSION

# conversion a valores
df['salary_in_usd'] = (df['salary_in_usd']).astype(int)

prom = df['salary_in_usd'].mean()
desvtd = df['salary_in_usd'].std()
var = df['salary_in_usd'].var()
quin = df['salary_in_usd'].quantile([0.2, 0.4, 0.6, 0.8, 1])
quin = pd.DataFrame({'Quintiles' : quin})
quin
max = df['salary_in_usd'].max()
min = df['salary_in_usd'].min()
rang = df['salary_in_usd'].max() - df['salary_in_usd'].min()

print(f"Promedio: {prom.round(2)}")
print(f"Desviacion Standard: {desvtd.round(2)}")
print(f"Varianza: {var.round(2)}")
print(f"Rango minimo: {min}, Rango maximo: {max}")
print(f"Rango: {rang}")
quin

Promedio: 137570.39
Desviacion Standard: 63055.63
Varianza: 3976011879.23
Rango minimo: 5132, Rango maximo: 450000
Rango: 444868


Quintiles
0.2    83972.8
0.4   120000.0
0.6   149850.0
0.8   185900.0
1.0   450000.0

2. Una comparación entre los salarios agrupados por 3 diferentes categorías a tu elección. ¿Para cuál de las medidas de tendencia central son más representativas? ¿Para cuál son menos representativas? Explica.

In [105]:
# Agrupacion por media aritmetica
Moda = df.groupby('salary_currency').agg(lambda x: x.mode().iloc[0])
ModaOr_Curr = Moda.sort_values(by='salary_in_usd', ascending=False).head(10)
ModaOr_Curr['salary_in_usd']

Promedio = df.groupby('company_location')['salary_in_usd'].mean()
PromOr_loca = Promedio.sort_values(ascending=False).round(2).head(10)

Mediana = df.groupby('company_size')['salary_in_usd'].median()
MedianaOr_Size = Mediana.sort_values(ascending=False).round(2)


In [106]:
print(f"Moda de Salario por moneda:\n{ModaOr_Curr['salary_in_usd']}\nEn orden de salarios que mas se repiten por cada moneda\n------------------------------------\n< MAS REPRESENTATIVO EN LOS DATOS >\nPorque dice del salario que mas se paga como tendencia a ganar por empleados segun la moneda osea depende del lugar de origen de la empresa.\nEn este caso es Israel")

Moda de Salario por moneda:
salary_currency
ILS    423834
USD    100000
GBP     92350
EUR     84053
AUD     83171
CAD     76814
SGD     75020
HKD     65062
CHF     56536
JPY     41689
Name: salary_in_usd, dtype: int64
En orden de salarios que mas se repiten por cada moneda
------------------------------------
< MAS REPRESENTATIVO EN LOS DATOS >
Porque dice del salario que mas se paga como tendencia a ganar por empleados segun la moneda osea depende del lugar de origen de la empresa.
En este caso es Israel


In [107]:
print(f"Promedio de salario por ubicacion:\n{PromOr_loca}\nEn orden de promedios de salarios por cada ubicacion\n------------------------------------\n< PROMEDIO MAS REPRESENTATIVO EN LOS DATOS >\nPorque dependiendo del lugar marca una tendencia afectada por datos de salarios muy bajos que podrian ser pocos.\nEn este caso es Israel coincidiendo con el top de la moda")

Promedio de salario por ubicacion:
company_location
IL    271446.50
PR    167500.00
US    151822.01
RU    140333.33
CA    131917.69
NZ    125000.00
BA    120000.00
IE    114943.43
JP    114127.33
SE    105000.00
Name: salary_in_usd, dtype: float64
En orden de promedios de salarios por cada ubicacion
------------------------------------
< PROMEDIO MAS REPRESENTATIVO EN LOS DATOS >
Porque dependiendo del lugar marca una tendencia afectada por datos de salarios muy bajos que podrian ser pocos.
En este caso es Israel coincidiendo con el top de la moda


In [108]:
print(f"Mediana de salario por tamaño de empresa:\n{MedianaOr_Size}\nEn orden de Medianas de salarios por cada tamaño\n------------------------------------\n< MEDIANA NO REPRESENTATIVO EN LOS DATOS >\nPorque no genera un dato relevante ante la cantidad de salarios arriba o abajo de la mediana, no genera una respuesta relevante respecto a salarios.")

Mediana de salario por tamaño de empresa:
company_size
M    140000.0
L    108500.0
S     62146.0
Name: salary_in_usd, dtype: float64
En orden de Medianas de salarios por cada tamaño
------------------------------------
< MEDIANA NO REPRESENTATIVO EN LOS DATOS >
Porque no genera un dato relevante ante la cantidad de salarios arriba o abajo de la mediana, no genera una respuesta relevante respecto a salarios.


3. Interpretar los indicadores estadísticos en cada caso, considerando medidas de tendencia central y dispersión.

In [109]:
# CALCULOS REPRESENTATIVOS
PromedioS = df.groupby('company_size')['salary_in_usd'].mean()
PromOr_Size = PromedioS.sort_values(ascending=False).round(2).head(10)

PromedioC = df.groupby('salary_currency')['salary_in_usd'].mean()
PromOr_Currency = PromedioC.sort_values(ascending=False).round(2).head(10)

print(f"\npromedio por tamaño: {PromOr_Size}")
print(f"\npromedio por moneda: {PromOr_Currency}")
print(f"\npromedio por ubicación: {PromOr_loca}")


promedio por tamaño: company_size
M    143130.55
L    118300.98
S     78226.68
Name: salary_in_usd, dtype: float64

promedio por moneda: salary_currency
ILS    423834.00
USD    149366.91
CHF    100682.00
CAD     96707.40
GBP     83850.23
SGD     81770.00
AUD     74198.44
HKD     65062.00
EUR     62281.73
JPY     60921.33
Name: salary_in_usd, dtype: float64

promedio por ubicación: company_location
IL    271446.50
PR    167500.00
US    151822.01
RU    140333.33
CA    131917.69
NZ    125000.00
BA    120000.00
IE    114943.43
JP    114127.33
SE    105000.00
Name: salary_in_usd, dtype: float64


RESPUESTA:
- La Desviacion Standard: 63055.63 de los salarios en usd de los datos es relativamente pequeña, eso quiere decir que la desviacion es menos del 30% del promedio, nos indica que la tendencia por pago de los salarios tiende a igualarse, osea que la desvicion baja nos dice que hay tambien una desvicion baja.

- El Rango: 444868 en salarios de usd es alto pero debido a la desvicion baja, se supone que los datos mas lejanos al promedio son relativamente pocos en comparacion a la concentracion de puntos.

- Promedio: 137570.39 en salarios en usd pero se centra mas con tendencia al limite inferior al del rango osea por debajo del percentil 0.6	(149850.0)

Quintiles

0.2	  83972.8

0.4	  120000.0

0.6	  149850.0

0.8	  185900.0

1.0	  450000.0




- Las empresas medianas tienen un pago en promedio de 143130.55 usd por encima de las empresas grandes.

- El pago en moneda mayor es ILS 423834.00 usd lo que puede entenderse que en Israel, puede estar mas valorado la misma labor que en otros paises.

- El pago por Ubicacion IL 271446.50 sigue liderando no solo economicamente la moneda israeli puede mostrarse mas fuerte si no que en Israel las empresas tienen mayor capacidad de pago a sus empleados 

- El salario que mas se repite es ILS 423834 lo que esta del segundo bastante distante, lo que indica que el mayor salario que mas pagan por la labor esta en Israel.


*CONCLUSION:
ISRAEL parece tener mayor estabilidad economica a nivel mundial por tener los empleos mejor pagados.

